viewing_party/party.py

```diff
@@ -1,23 +1,190 @@
+    for genre, count in genre_frequency.items():
+        if count > highest_genre_count:
+            highest_genre_count = count
+            most_watched_genre = genre
+
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    unique_movies_list = []
+    total_friends_watched = []
+
+    #loops through values of key 'friends' in user data, and is appended to total_friends_watched list
+    for movies in user_data["friends"]:
+        total_friends_watched += movies["watched"]
+
+    #loops through values of key 'watched' in user_data to get movies watched
+    for unique_movies in user_data["watched"]:
+        #if statement compares movies in total_friends_watched list to what the user has watched, any unique movie gets appended to unique_movies_list
+        if unique_movies not in total_friends_watched:
+            unique_movies_list.append(unique_movies)
+    
+    return unique_movies_list
+
+
+def get_friends_unique_watched(user_data):
+    unique_movies_list = []
+    total_friends_watched = []
+
+    #loops through values of key 'friends' in user data, and is appended to total_friends_watched list
+    for movies in user_data["friends"]:
+        total_friends_watched += movies["watched"]
+    
+    #loops through movies in the total_friends_watched list to compare (in the if statement) if the movie is not in the user's watched list and hasn't appeared in the unique_movies_list (checks for duplicates) to then append the unique movie
+    for unique_movies in total_friends_watched:
+        if (unique_movies not in user_data["watched"]) and (not unique_movies in unique_movies_list):
+            unique_movies_list.append(unique_movies)
+    
+    return unique_movies_list
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    # list titles of movies the user has already watched 
+    user_watched_list = []
+    for movie in user_data["watched"]:
+        user_watched_list.append(movie["title"])
+    
+    # list that has rec movies
+    recommendations = []
+
+    # loop through each friend
+    for friend in user_data["friends"]:
+        # loop through their moviees
+        for friend_movie in friend["watched"]:
+            # check if user hasn't watched the movie and if host (streaming services) is also under user
+            if (friend_movie["title"] not in user_watched_list and friend_movie["host"] in user_data["subscriptions"] and friend_movie not in recommendations):
+                recommendations.append(friend_movie)
+
+    return recommendations
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):

```

Comment:
`get_new_rec_by_genre` is duplicating work that we've already done to find the most common genre for a user and only look at movies that at least one friend has seen, but we have not. How can we reuse our previous functions `get_most_watched_genre` and `get_friends_unique_watched` to simplify our approach here?